azurerm/internal/services/compute/windows_virtual_machine_scale_set_resource.go

```diff
@@ -969,6 +984,18 @@ func resourceArmWindowsVirtualMachineScaleSetRead(d *schema.ResourceData, meta i
 				return fmt.Errorf("Error setting `terminate_notification`: %+v", err)
 			}
 		}
+
+		if features.VMSSExtensionsBeta() {
+			if profile.ExtensionProfile != nil {
+				if extensionProfile, err := flattenVirtualMachineScaleSetExtensions(profile.ExtensionProfile, d); err != nil {
+					return fmt.Errorf("failed flettening `vm_extension`: %+v", err)
+				} else if err := d.Set("extension", extensionProfile); err != nil {
+					return fmt.Errorf("failed to set vm_extension: %+v", err)
+				}
+			} else {
+				d.Set("extension", []map[string]interface{}{})
+			}
```

Comment:
(as above) we can remove if/else statement here - since the flatten function should return an empty list if it's nil